### ISSUE TITLE
kubectl config not so user-friendly

### ISSUE THREAD

**Issue by jlowdermilk** [GoogleCloudPlatform/kubernetes#3728]
Tracking friction points encoutered using ```kubectl config``` commands

```kubectl config set-cluster``` usage string is incorrect; indicates values for server, certificate-authority etc, should be passed as args, but command actually takes them as flags.

Difficult to see which flags are actually important for the command. Default cobra help prints all available flags in one long list, including unrelated/top-level flags. Usage string should include the important flags for ```set-cluster|user|context```

```kubectl config set-cluster|user|context``` should not wipe existing values if a new one is not specified. e.g.
```
kubectl config set-cluster e2e --certificate-authority=~/.kube/e2e/.kubernetes.ca.cert
```
should just set the `certificate-authority` field on the e2e cluster entry without wiping server,api-version.

/cc @deads2k

**Comment by deads2k** [comment]
Thanks, I'll take a look this afternoon.

**Comment by goltermann** [comment]
Jeff, can you close if this is done?

**Comment by jlowdermilk** [comment]
Closed by #3768
